Sourcery/Parsing/Composer.swift

```diff
@@ -91,13 +91,74 @@ struct Composer {
         return unique.values.sorted { $0.name < $1.name }
     }
 
+    private func resolveGenerics(typeName: TypeName, unique: [String: Type]) {
+        let genericSplit = typeName.unwrappedTypeName.characters.split(separator: "<", maxSplits: 1)
+
+        guard genericSplit.count == 2 else {
+            return
+        }
+
+        let typeCharacters = genericSplit[0]
+        var genericCharacters = genericSplit[1]
+
+        guard genericCharacters.count > 1, genericCharacters.removeLast() == ">" else {
+            return
+        }
+
+        var genericTypeStrings = [String]()
+
+        var subGenericsCounter = 0
+        var characterBuffer = [Character]()
+
+        for character in genericCharacters {
```

Comment:
Oh, sorry! I'm used to writing performant parsers for formats like JSON, BSON and the like. I naturally keep performance as much as possible in the back of my mind as a habit and already gave in to using Strings rather than bytes 😂